src/Log.ts

```diff
@@ -47,76 +71,96 @@ export interface ILog {
      */
     (...args: unknown[]): Log;
     /**
-     * Intercept, redefine or redirect any log
+     * Intercept,redefine or redirect any log
      * If you clear args you intercept the log and nothing happen more after this call.
      * @param type log level
      * @param args args
      * @returns
      */
-    on: (type: LogType, args: unknown[]) => void;
+    on: (level: LogLevel, args: unknown[]) => void;
+    /**
+     * Change log level, default log level is {@link LogLevel.INFO},
+     * or {@link LogLevel.DEBUG} when there is an intentional 'debug' query parameter in the loaded page
+     * If null it interceps all the logs
+     */
+    level?: LogLevel | null;
 }
 
+// check coder issuer: everytime we don't forget to use the built Log
 let _logging = 0;
 setInterval(() => {
     console.assert(_logging === 0, _logging.toFixed(), 'calls to log was useless');
 }, 10000);
 
-/**
- * Log types
- */
-export enum LogType {
-    ERROR = 'error',
-    WARN = 'warn',
-    INFO = 'info',
-    DEBUG = 'debug'
-}
+const _defaulLevel =
+    typeof location !== 'undefined' && location.search.startsWith('?debug!') ? LogLevel.DEBUG : LogLevel.INFO;
```

Comment:
I think using `window.location` is a little bit intrusive for widely distributed JS library.
We should at least document this behavior in the README file like this : 

```
Debug Mode Notice:
If the main page’s URL query string begins with ?debug! (for example, https://example.com?debug!), this library will automatically set the global log level to debug. This is intended to help with troubleshooting and diagnostics during development or testing.

Important:

This feature can be triggered by anyone who can alter the URL, so use it with caution in production environments.
For details on log levels, see LogLevel Documentation.
```

Reply:
Agree!!! Good finding @thomasjammet 
We should remove that

**Environment Coupling**: Relying on location.search ties your library’s logic directly to a browser environment. This makes the library harder to use or test in non-browser contexts (e.g., Node.js, server-side rendering, or test frameworks) since location may not be defined there. It reduces the portability and reusability of your code.

**Hidden Configuration**: Using query parameters to control behavior introduces a hidden and potentially undocumented configuration path. Instead of having a clear configuration API or passing an explicit setting into the library, you’re relying on the runtime URL state. This makes it less transparent for consumers of the library and harder for future maintainers to understand where and how the log level is being set.

**Limited Flexibility & Inconsistent Behavior**: Depending on query parameters to determine behavior is brittle and may break if the URL structure changes or if the application is embedded in contexts that can’t manipulate query strings easily. If the logging level is supposed to be consistent across different environments or deployment scenarios, relying on a query param provides a single, external trigger point rather than a properly managed configuration interface, leading to inconsistent or unreliable behavior.

Reply:
[refactor(log): implement !cb-override-log-level, also add true to enable all, false to disable all](https://github.com/CeeblueTV/web-utils/commit/db83c03ff1a4e334d102408889804e0e98e4e42b)

I have called it `!cb-override-log-level` for:
- `!cb` => special 3 letter unique for us
- `override-log-level` => the naming function

I hope that it makes sense! Thank you for helping me to find the best ever solution 🚀